manifests-maistra/1.2.0/servicemeshcontrolplanes.crd.yaml

```diff
@@ -101,6 +101,8935 @@ spec:
             description: The current status of this ServiceMeshControlPlane and the
               components that comprise the control plane. This data may be out of
               date by some window of time.
+            nullable: true
+            properties:
+              annotations:
+                additionalProperties:
+                  type: string
+                description: Annotations is an unstructured key value map used to
+                  store additional, usually redundant status information, such as
+                  the number of components deployed by the ServiceMeshControlPlane
+                  (number is redundant because you could just as easily count the
+                  elements in the ComponentStatus array). The reason to add this redundant
+                  information is to make it available to kubectl, which does not yet
+                  allow counting objects in JSONPath expressions.
+                type: object
+              components:
+                items:
+                  description: ComponentStatus represents the status of an object
+                    with children
+                  properties:
+                    children:
+                      description: 'TODO: can we remove this? it''s not used anywhere
```

Comment:
Are you planning on removing hte TODO before merge?

Reply:
I anticipate a couple of rounds of cleanup, so not at this point.  I need to get this in so we can start getting 2.0 installations going.  This is a pretty big change and needs a lot of eyeballs and a proper design review.  I don't anticipate this being the final revision, but a starting point to get it dialed in.